tests/rptest/services/redpanda.py

```diff
@@ -3820,50 +3898,128 @@ def all_partitions_uploaded_manifest():
+        """
+        Configure the scrubber such that it will run aggresively
+        until the entire partition is scrubbed. Once that happens,
+        the scrubber will pause due to the `cloud_storage_full_scrub_interval_ms`
+        config. Returns the aggregated anomalies for all partitions after applying
+        filtering for expected damage.
+        """
+        if not cloud_storage_partitions:
+            return None
 
-        # It is legal for tiered storage to leak objects under
-        # certain circumstances: this will remain the case until
-        # we implement background scrub + modify test code to
-        # insist on Redpanda completing its own scrub before
-        # we externally validate
-        # (https://github.com/redpanda-data/redpanda/issues/9072)
-        permitted_anomalies = {"segments_outside_manifest"}
+        self.set_cluster_config({
+            "cloud_storage_partial_scrub_interval_ms":
+            100,
+            "cloud_storage_full_scrub_interval_ms":
+            1000 * 60 * 10,
+            "cloud_storage_scrubbing_interval_jitter_ms":
+            100,
+            "cloud_storage_background_jobs_quota":
+            5000,
+            "cloud_storage_housekeeping_interval_ms":
+            100,
+            # Segment merging may resolve gaps in the log, so disable it
+            "cloud_storage_enable_segment_merging":
+            False,
+            # Leadership moves may perturb the scrub, so disable it to
+            # streamline the actions below.
+            "enable_leader_balancer":
+            False
+        })
+
+        unavailable = set()
+        for p in cloud_storage_partitions:
+            try:
+                leader_id = self._admin.await_stable_leader(topic=p.topic,
+                                                            partition=p.index)
+
+                self._admin.reset_scrubbing_metadata(
+                    namespace="kafka",
+                    topic=p.topic,
+                    partition=p.index,
+                    node=self.get_node(leader_id))
+            except HTTPError as he:
+                if he.response.status_code == 404:
+                    # Old redpanda, doesn't have this endpoint.  We can't
+                    # do our upload check.
+                    unavailable.add(p)
+                    continue
+                else:
+                    raise
+
+        cloud_storage_partitions -= unavailable
+        scrubbed = set()
+        all_anomalies = []
+
+        allowed_keys = set([
+            "ns", "topic", "partition", "revision_id", "last_complete_scrub_at"
+        ])
+
+        expected_damage = self._si_settings.get_expected_damage()
+
+        def filter_anomalies(detected):
+            bad_delta_types = set(
+                ["non_monotonical_delta", "mising_delta", "end_delta_smaller"])
+
+            for anomaly_type in expected_damage:
+                if anomaly_type == "ntpr_no_manifest":
```

Comment:
nit: worth adding a comment that this (i think) is translating rp-storage-tool anomaly types to scrubber types, hence the difference in naming?

Reply:
Yeah. Since we have two scrubbers that report anomalies differently, we have to use a single format for exemptions. I chose to use the existing format instead of creating a new one. I'll add a comment here.

Reply:
Done.